app/migrations/versions/d0088b4df549_.py

```diff
@@ -0,0 +1,30 @@
+"""empty message
+
+Revision ID: d0088b4df549
+Revises: b40055ad9587
+Create Date: 2019-10-30 22:15:22.015323
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = 'd0088b4df549'
+down_revision = 'b40055ad9587'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('user', sa.Column('_encrypted_pin', sa.String(length=128), nullable=True))
+    op.add_column('user', sa.Column('failed_pin_attempts', sa.Integer(), nullable=True))
```

Comment:
hm, do defaults not register in `add_column`?

Reply:
Just checked up on this and found that it does present an error.
So weed need to run some SQL there before setting nullable to false but I am not so sure how clean its is.
It would end up being something like:
`
op.add_column(
        'users', 
        sa.Column('failed_pin_attempts', sa.Integer(), nullable=True)
    )
    op.execute("UPDATE users SET failed_pin_attempts = false")
    op.alter_column('users', 'failed_pin_attempts', nullable=False)
`

Reply:
https://www.programcreek.com/python/example/87970/alembic.op.add_column
looks like you can pass in a default. don't know why the auto-generated migration didn't have it but should be able to modify and add it in